pkg/runtime/framework/plugins/mpi/mpi.go

```diff
@@ -176,90 +216,96 @@ func (m *MPI) ReconcilerBuilders() []runtime.ReconcilerBuilder {
 	}
 }
 
-func (m *MPI) Build(_ context.Context, info *runtime.Info, trainJob *trainer.TrainJob) ([]any, error) {
+func (m *MPI) Build(ctx context.Context, info *runtime.Info, trainJob *trainer.TrainJob) ([]any, error) {
 	if info == nil || info.RuntimePolicy.MLPolicy == nil || info.RuntimePolicy.MLPolicy.MPI == nil {
 		return nil, nil
 	}
 
-	secret, err := m.buildSSHAuthSecret(trainJob)
-	if err != nil {
-		return nil, fmt.Errorf("failed to build Secret with SSH auth keys. Error: %v", err)
-	}
+	var objects []any
 
+	// SSHAuthSecret is immutable.
+	if err := m.client.Get(ctx, client.ObjectKey{Name: sshAuthSecretName(trainJob.Name), Namespace: trainJob.Namespace}, &corev1.Secret{}); err != nil {
```

Comment:
Why do you need to get secret before creating it ?

Reply:
Because Secret is an immutable object. If there is Secret, already, we should not return Secret. If we return the Secret regardless of its secret existence, TrainJob controllers will fail to PATCH the operation.

Reply:
Ain't Secrets and CM immutable only if we enable it via APIs: https://kubernetes.io/docs/concepts/configuration/secret/#secret-immutable ?

Reply:
Yes, as you can see in the Secret construction implementation, we specify `immutable`.